config/CMakeAddFortranSubdirectory.cmake

```diff
@@ -518,14 +641,190 @@ function( cafs_create_imported_targets targetName libName targetPath linkLang)
       set_target_properties(${targetName}
         PROPERTIES
         IMPORTED_IMPLIB_DEBUG
-        "${libloc_debug}/${CMAKE_IMPORT_LIBRARY_PREFIX}${libName}${CMAKE_IMPORT_LIBRARY_SUFFIX}"
+          "${libloc_debug}/${CMAKE_IMPORT_LIBRARY_PREFIX}${libName}${CMAKE_IMPORT_LIBRARY_SUFFIX}"
         )
     endif()
   endif()
   unset(lib CACHE)
   unset(lib_debug CACHE)
 endfunction()
 
-#-------------------------------------------------------------------------------#
+#------------------------------------------------------------------------------#
+# Generate a set of default variables for the CAFS cmake command line.
+# This macro sets these cmake variables:
+#   - build_system_state
+# Optional arguments:
+# - PROJECT "Draco"
+#------------------------------------------------------------------------------#
+function(init_build_system_state)
```

Comment:
Okay, just so I understand-- this guy gets called in the CMakeLists.txt for all packages that have Fortran code in them... and its purpose is largely to sort out if the build system needs additional variables to be set to properly build and install the Fortran-dependent parts of draco. Is that generally correct?

(Thank you for commenting this code well so I could kind of follow it, despite my general lack of CMake wisdom :D )

Reply:
* This is the hardest cmake I've ever dealt with.   Feel free to skip over some of the CAFS details.
* This function is only called for Fortran subdirectories when the master build project doesn't know what Fortran is (i.e. MSVC or Xcode).  In these cases, the Fortran directories become independent Makefile-based projects built as a sub-shell operation.

Reply:
Ahh, okay! Makes sense. Thanks! :)